### ISSUE TITLE
authHandler examples

### ISSUE THREAD

**Issue by antiochtech** [mscdex/ssh2#1020]
can we get some?

sorry to ask here, but the question just gets crapped on at stackoverflow.

**Comment by antiochtech** [comment]
many thanks to Riddell at SO for the following!
```
// Require Client Class from ssh2
const { Client } = require('ssh2');

// Create instance of Client (aka connection)
const conn = new Client();

// Create our ready event that's called after we connect
conn.on('ready', () => {
    console.log('Client :: ready');
});

// Connect with a config object passed in the parameters
conn.connect({
    host: '192.168.100.100',
    port: 22, // SSH

    // Authentication Handler
    authHandler: function (methodsLeft, partialSuccess, callback) {

        // Code e.g. get credentials from database
    
        // Once your logic is complete invoke the callback
        // http://npmjs.com/package/ssh2#client-examples
        callback({
            type: 'password',
            username: 'foo',
            password: 'bar',
        });
    }
});
```
or
```
conn.on('ready', () => {
    console.log('Client :: ready');
}).connect({ // Chained
    host: '192.168.100.100',
    port: 22, // SSH

    // Authentication Handler
    authHandler: function (methodsLeft, partialSuccess, callback) {

        // Code e.g. get credentials from database
    
        // Once your logic is complete invoke the callback
        // http://npmjs.com/package/ssh2#client-examples
        callback({
            type: 'password',
            username: 'foo',
            password: 'bar',
        });
    }
});
```

**Comment by mscdex** [comment]
FWIW you don't have to even define a function if you already have the auth information upfront, you can just set `authHandler` to an array of the kinds of objects the function would provide:

```js
{
  // ...
  authHandler: [
    { type: 'password', username: 'foo', password: 'bar' },
  ],
}
```

The use of the function is really only necessary if you are doing something more involved and need to perform some dynamic logic (e.g. fetching a username and/or password from a remote server).

**Comment by antiochtech** [comment]
couldve sworn i tried exactly that before posting. but im having another issue that mightve confused the situation when i did. hope to get it figured out without posting. in any case, many thanks!

**Comment by mscdex** [comment]
Also, you will need v1.0.0+ for this functionality.
